src/solver/simulation/sim_structure_donnees.h

```diff
@@ -64,6 +64,8 @@ typedef struct
     double* ParLigne;
 } MATRICE_2D;
 
+typedef std::vector<std::vector<VALEURS_GENEREES_PAR_PAYS>> VAL_GEN_PAR_PAYS;
```

Comment:
Possibly for a future PR but:
the best thing would be to have a small class to make the indices more clear. Here we don't really know what is the index in the first vector, and in the second vector.

We could define something like
```cpp
// All generated values for one space
class SpaceGeneratedValues {
public:
    VALEURS_GENEREES_PAR_PAYS& areaValues(int areaIndex) {
        return values[areaIndex];
    }
    // const overload

private:
   std::vector<VALEURS_GENEREES_PAR_PAYS> values;
};

// All generated values for all spaces, replaces the typedef
class GeneratedValues {
public:
    SpaceGeneratedValues& spaceValues(int spaceIndex) {
        return values[areaIndex];
    }
    // const overload
private:
   std::vector<SpaceGeneratedValues> values;
};

```

Reply:
I agree, with vectors of vectors it's not clear what index is for what : area ? numSpace ?